dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/service/AnonymisingDataService.java

```diff
@@ -0,0 +1,281 @@
+        if (hasValue(applicant.getFirstName())) {
+            applicant.setFirstName(faker.name().firstName());
+        }
+        if (hasValue(applicant.getLastName())) {
+            applicant.setLastName(faker.name().lastName());
+        }
+        Optional.ofNullable(applicant.getDob()).ifPresent(dob ->
+                applicant.setDob(convertToXMLGregorianCalendar(faker.timeAndDate().birthday()))
+        );
+        if (hasValue(applicant.getNiNumber())) {
+            applicant.setNiNumber(faker.idNumber().valid());
+        }
+        if (hasValue(applicant.getLandline())) {
+            applicant.setLandline(faker.phoneNumber().phoneNumber());
+        }
+        if (hasValue(applicant.getMobile())) {
+            applicant.setMobile(faker.phoneNumber().cellPhone());
+        }
+        if (hasValue(applicant.getEmail())) {
+            applicant.setEmail(faker.internet().emailAddress());
+        }
+        if (hasValue(applicant.getSpecialInvestigation())) {
+            applicant.setSpecialInvestigation(faker.lorem().sentence());
+        }
+
+        Optional.ofNullable(applicant.getPreferredPaymentMethod()).ifPresent(paymentMethod -> {
+            if (hasValue(paymentMethod.getCode())) {
+                paymentMethod.setCode(faker.text().text(4));
+            }
+            if (hasValue(paymentMethod.getDescription())) {
+                paymentMethod.setDescription(faker.text().text(10));
+            }
+        });
+    }
+
+    private void anonymiseBankDetails(CONTRIBUTIONS.Applicant.BankDetails bankDetails) {
+        Optional.ofNullable(bankDetails).ifPresent(details -> {
+            if (hasValue(details.getAccountName())) {
+                details.setAccountName(faker.name().fullName());
+            }
+            if (hasValue(details.getSortCode())) {
+                details.setSortCode(faker.number().numberBetween(100000, 999999));
+            }
+            if (hasValue(details.getAccountNo())) {
+                details.setAccountNo(faker.number().numberBetween(10000000, 99999999));
+            }
+        });
+    }
+
+    private void anonymisePartnerPersonalDetails(CONTRIBUTIONS.Applicant.PartnerDetails partnerDetails) {
+        Optional.ofNullable(partnerDetails).ifPresent(details -> {
+            if (hasValue(details.getFirstName())) {
+                details.setFirstName(faker.name().firstName());
+            }
+            if (hasValue(details.getLastName())) {
+                details.setLastName(faker.name().lastName());
+            }
+            if (hasValue(details.getNiNumber())) {
+                details.setNiNumber(faker.idNumber().valid());
+            }
+            Optional.ofNullable(details.getDob()).ifPresent(dob ->
+                    details.setDob(convertToXMLGregorianCalendar(faker.timeAndDate().birthday()))
+            );
+        });
+    }
+
+    private void anonymisePartnerCirDetails(CONTRIBUTIONS.Applicant.Partner partner) {
```

Comment:
same thing here - I know the confluence page says to anonymise this but it shouldn't be. Can you remove it from the code ? I will update the confluence page. this comment is for the PARTNER->CONTRARY INTEREST DETAILS